pkg/kubernetes/podtemplatespec/v1alpha1/podtemplatespec.go

```diff
@@ -0,0 +1,471 @@
+
+	// copy of original pointer
+	newaffinitylist := *affinity
+
+	b.podtemplatespec.Object.Spec.Affinity = &newaffinitylist
+	return b
+}
+
+// WithTolerations merges the existing tolerations
+// with the provided arguments
+func (b *Builder) WithTolerations(tolerations ...corev1.Toleration) *Builder {
+	if tolerations == nil {
+		b.errs = append(
+			b.errs,
+			errors.New(
+				"failed to build podtemplatespec object: nil tolerations",
+			),
+		)
+		return b
+	}
+	if len(tolerations) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New(
+				"failed to build podtemplatespec object: missing tolerations",
+			),
+		)
+		return b
+	}
+
+	if len(b.podtemplatespec.Object.Spec.Tolerations) == 0 {
+		return b.WithTolerationsNew(tolerations...)
+	}
+
+	b.podtemplatespec.Object.Spec.Tolerations = append(
+		b.podtemplatespec.Object.Spec.Tolerations,
+		tolerations...,
+	)
+
+	return b
+}
+
+// WithTolerationsNew sets the tolerations field of podtemplatespec
+func (b *Builder) WithTolerationsNew(tolerations ...corev1.Toleration) *Builder {
+	if tolerations == nil {
+		b.errs = append(
+			b.errs,
+			errors.New(
+				"failed to build podtemplatespec object: nil tolerations",
+			),
+		)
+		return b
+	}
+	if len(tolerations) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New(
+				"failed to build podtemplatespec object: missing tolerations",
+			),
+		)
+		return b
+	}
+
+	// copy of original slice
+	newtolerations := []corev1.Toleration{}
+	newtolerations = append(newtolerations, tolerations...)
+
+	b.podtemplatespec.Object.Spec.Tolerations = newtolerations
+
+	return b
+}
+
+// WithContainerBuilders builds the list of containerbuilder
+// provided and merges it to the containers field of the podtemplatespec
+func (b *Builder) WithContainerBuilders(
+	containerBuilderList ...*container.Builder,
+) *Builder {
+	if containerBuilderList == nil {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build podtemplatespec: nil containerbuilder"),
+		)
+	}
+	for _, containerBuilder := range containerBuilderList {
+		containerObj, err := containerBuilder.Build()
+		if err != nil {
+			b.errs = append(
+				b.errs,
+				errors.Wrap(
+					err,
+					"failed to build podtemplatespec",
+				),
+			)
+			return b
+		}
+		b.podtemplatespec.Object.Spec.Containers = append(
+			b.podtemplatespec.Object.Spec.Containers,
+			containerObj,
+		)
+	}
+	return b
+}
+
+// WithVolumeBuilders builds the list of volumebuilders provided
+// and merges it to the volumes field of podtemplatespec.
+func (b *Builder) WithVolumeBuilders(
+	volumeBuilderList ...*volume.Builder,
+) *Builder {
+	if volumeBuilderList == nil {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build podtemplatespec: nil volumeBuilderList"),
+		)
+	}
+	for _, volumeBuilder := range volumeBuilderList {
+		vol, err := volumeBuilder.Build()
+		if err != nil {
+			b.errs = append(
+				b.errs,
+				errors.Wrap(err, "failed to build podtemplatespec"),
+			)
+			return b
+		}
+		newvol := *vol
```

Comment:
why '*' here when other one for containers doesn't need?

Reply:
It is because the `Build()` function of container returns `corev1.Container` and the `Build()` function of volume returns `*corev1.Volume`